File: backend/schemas/memory.py
from typing import Optional
from pydantic import BaseModel, Field
from datetime import datetime


class MemoryBase(BaseModel):
    user_id: str
    key: str
    value: str
    context: Optional[str] = None
    priority: float = Field(default=0.5, ge=0.0, le=1.0)


class MemoryCreate(MemoryBase):
    id: Optional[str] = None


class MemoryResponse(MemoryBase):
    id: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/memory.ts
export interface MemoryBase {
  user_id: string;
  key: string;
  value: string;
  context?: string;
  priority: number; // 0-1
}

export interface MemoryCreate extends MemoryBase {
  id?: string;
}

export interface MemoryResponse extends MemoryBase {
  id: string;
  created_at: string;
  updated_at: string;
}